### PR TITLE
Numpy Deprecated Code Replacement

### PR DESCRIPTION
**Pull request by shubham1206agra**
- Disabled use of Deprecated Numpy API use by defining kwarg in Extension in `setup.py.in` 
```py
define_macros=[('NPY_NO_DEPRECATED_API', 'NPY_1_7_API_VERSION')]
```
which will turn it in c++ into
```cpp
#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
```
- Changing direct use of members of `PyArrayObject` to function based access
1. `X.data` changes to `PyArray_DATA(X)`
2. `X.shape` changes to `PyArray_SHAPE(X)`
3. `X.flags` changes to `PyArray_FLAGS(X)`

- Changed `NPY_OWNDATA` to `NPY_ARRAY_OWNDATA`
- Changed the if logic
```py
if not X.flags.c_contiguous or (not X.flags.owndata and not isWin)
```
to
```py
if not (flags & numpy.NPY_ARRAY_C_CONTIGUOUS) or (not (flags & numpy.NPY_ARRAY_OWNDATA) and not isWin)
```

### CODE REVIEW

**Review by rcurtin — COMMENTED** [review]
This is awesome!  Thank you so much for taking the time to fix this issue.  I really like little code maintenance PRs like this. :100: I tested the build myself and indeed, no more warnings. :rocket: 

I only have a couple little comments relating to style, and then we should definitely merge this. :+1:

File: src/mlpack/bindings/python/print_input_processing.hpp [hunk 1]
```diff
@@ -534,4 +540,4 @@ void PrintInputProcessing(util::ParamData& d,
 } // namespace bindings
 } // namespace mlpack
 
-#endif
+#endif
```
[the remark this answers is not in the document]
Comment (shubham1206agra) [reply src/mlpack/bindings/python/print_input_processing.hpp:543]: I haven't removed it.
And I am unable to put it back.

[the remark this answers is not in the document]
Comment (shubham1206agra) [reply src/mlpack/bindings/python/print_input_processing.hpp:543]: It's done by vscode automatically.

File: src/mlpack/bindings/python/mlpack/arma_numpy.pyx [hunk 2]
```diff
@@ -48,17 +51,18 @@ cdef arma.Mat[double]* numpy_to_mat_d(numpy.ndarray[numpy.double_t, ndim=2] X, \
   """
   Convert a numpy ndarray to a matrix.  The memory will still be owned by numpy.
   """
-  if not X.flags.c_contiguous or (not X.flags.owndata and not isWin):
+  cdef int flags = PyArray_FLAGS(X)
+  if not (flags & numpy.NPY_ARRAY_C_CONTIGUOUS) or (not (flags & numpy.NPY_ARRAY_OWNDATA) and not isWin):
```
[the remark this answers is not in the document]
Comment (shubham1206agra) [reply src/mlpack/bindings/python/mlpack/arma_numpy.pyx]: Done

File: src/mlpack/bindings/python/print_input_processing.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (rcurtin) [reply src/mlpack/bindings/python/print_input_processing.hpp:544]: You can change your VSCode settings :)

[the remark this answers is not in the document]
Comment (shubham1206agra) [reply src/mlpack/bindings/python/print_input_processing.hpp]: Done

**Comment by shubham1206agra** [comment]
You may merge now.

**Review by rcurtin — APPROVED** [review]
Thanks!  I really appreciate this PR.  Would you like to add an entry to `CHANGELOG.md`? :+1:

**Comment by rcurtin** [comment]
Thanks @shubham1206agra!

**Pull request merged**